### DISCUSSION TITLE
Volume cache initialisation breaks official Postgres image

### DISCUSSION THREAD

**Discussion by ineffyble** [batect/batect#1000] [Q&A] [answered]
Trying to use a normal Docker volume with the `postgres:13` image from Docker Hub fails in batect, because batect's cache initialisation creates a dummy file in the volume, and the container refuses to start if the data directory is not empty.

There doesn't seem to be any easy way to turn off the cache initialisation, only override it if I have my own image.

The recommendation from postgres is to set the `PGDATA` variable to a subdirectory of the volume so it can be created. Unfortunately that seems to produce permissions issues and instead I get an access denied error during startup - there's some [interesting behaviour](https://github.com/docker-library/postgres/blob/master/docker-entrypoint.sh) in the entrypoint around permissions and swapping users that may be a factor.

I haven't been able to find any examples of using batect with Postgres that might shine some light on the best approach.

Running on Windows 10 with WSL2 backend.

**Comment by charleskorn** [comment, marked as answer]
Two questions so I can understand what's going on more:

* Are you trying to persist data between task runs for the Postgres DB? Or would it be acceptable for the DB to start afresh every time the container starts?
* Are you using run as current user mode for the Postgres container?

**Reply by ineffyble** [reply]
Yes, trying to persist data. If I mount a volume from a local path it works correctly. Problem there is that I hit the massive performance issues from trying to use files in my Windows FS via WSL/Docker.

Haven't explicitly set "run as current user" but docs suggest that's irrelevant for Windows where it always runs as current user (although I wonder if that's only applicable for Hyper-V backend and not WSL2?).

**Reply by charleskorn** [reply]
I've found a way to solve this problem, and generally make working with caches a bit faster as well - this will be released as part of v0.74.0 shortly.

Batect will no longer create the `.cache-init` file in cache volumes the first time the volume is used, which should solve this issue. You will need to remove the existing cache volume by running `./batect --clean` to get a fresh volume.

Please give it a go and let me know if you run into any more issues.

**Reply by ineffyble** [reply]
@charleskorn worked perfectly!
